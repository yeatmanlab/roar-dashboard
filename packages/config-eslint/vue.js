import { config as baseConfig } from "./index.js";
import vue from 'eslint-plugin-vue';
import cypress from 'eslint-plugin-cypress';
import globals from 'globals';

export const config = [
  ...baseConfig,
  ...vue.configs['flat/recommended'],

  // General JS
  {
    files: ['**/*.js'],
    languageOptions: {
      ecmaVersion: 'latest',
      sourceType: 'module',
      globals: {
        ...globals.browser,
        process: 'readonly'
      }
    }
  },

  // Vue-specific config
  {
    files: ['**/*.vue'],
    plugins: { vue },
    languageOptions: {
      ecmaVersion: 'latest',
      sourceType: 'module',
      globals: {
        ...globals.browser,
        process: 'readonly' // @TODO: Remove and fix code.
      }
    },
    rules: {
      // TODO: Re-enable and fix violations of the following rules:
      'vue/max-attributes-per-line': 'off',
      'vue/html-indent': 'off',
      'vue/html-self-closing': 'off',
      'vue/html-closing-bracket-newline': 'off',
      'vue/singleline-html-element-content-newline': 'off',
      'vue/multiline-html-element-content-newline': 'off',
      // @TODO: End.

      'vue/multi-word-component-names': 'off',
      'vue/no-undef-components': [
        'error',
        {
          ignorePatterns: [
            'AppSpinner',
            'GMapAutocomplete', // vue-google-maps-community-fork
            'i18n-t', // vue-i18n
            'router-link', // vue-router
            'router-view', // vue-router
          ],
        },
      ],
    },
  },

  // Cypress-specific config
  cypress.configs.recommended,
  {
    files: ['**/*.cy.js', '**/cypress/**/*.js'],
    plugins: { cypress },
    languageOptions: {
      ecmaVersion: 'latest',
      sourceType: 'module',
    }
  }
];