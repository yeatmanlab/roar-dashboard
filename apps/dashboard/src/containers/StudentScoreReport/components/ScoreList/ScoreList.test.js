import { describe, it, expect, beforeEach, vi } from 'vitest';
import { mount } from '@vue/test-utils';
import { createI18n } from 'vue-i18n';
import { ref } from 'vue';

// Mock the ScoreCard component to avoid Chart.js dependencies
vi.mock('./ScoreCard', () => ({
  default: {
    name: 'ScoreCard',
    template: '<div class="mocked-score-card">ScoreCard</div>',
    props: [
      'publicName',
      'scoreLabel',
      'score',
      'tags',
      'valueTemplate',
      'scoreToDisplay',
      'studentFirstName',
      'description',
      'scoresArray',
      'expanded',
      'longitudinalData',
      'taskId',
      'studentGrade',
    ],
  },
  ScoreCardScreen: {
    name: 'ScoreCard',
    template: '<div class="mocked-score-card">ScoreCard</div>',
    props: [
      'publicName',
      'scoreLabel',
      'score',
      'tags',
      'valueTemplate',
      'scoreToDisplay',
      'studentFirstName',
      'description',
      'scoresArray',
      'expanded',
      'longitudinalData',
      'taskId',
      'studentGrade',
    ],
  },
}));

// Mock the composable - needs to return computed refs or objects that behave like them
vi.mock('./useScoreListData', () => ({
  useScoreListData: vi.fn(() => {
    const mockTask = {
      taskId: 'swr',
      scoreToDisplay: 'percentileScore',
      percentileScore: {
        name: 'Percentile Score',
        value: 75,
        min: 0,
        max: 100,
        supportColor: '#4CAF50',
      },
      tags: [
        {
          icon: 'pi pi-check-circle',
          value: 'Required',
          severity: 'info',
          tooltip: 'Required assessment',
        },
      ],
      historicalScores: [],
    };

    // Return refs that will be auto-unwrapped in templates
    return {
      computedTaskData: ref([mockTask]),
      scoreValueTemplate: ref(() => '{value}%'),
      getTaskDescription: ref(() => ({
        keypath: 'scoreReports.percentileTaskDescription',
        slots: {
          percentile: '75th percentile',
          taskName: 'Single Word Reading',
          taskDescription: 'measures reading ability',
          supportCategory: 'Achieved Skill',
        },
      })),
      getTaskScoresArray: ref(() => [
        ['Raw Score', 50, 0, 100],
        ['Percentile', 75, 0, 100],
      ]),
    };
  }),
}));

// Import after mocks
const ScoreList = await import('./ScoreList.vue').then((m) => m.default);

describe('ScoreList.vue', () => {
  let i18n;

  const defaultProps = {
    studentFirstName: 'John',
    studentGrade: '3',
    taskData: {
      swr: {
        taskId: 'swr',
        scores: {
          composite: {
            rawScore: 50,
            percentileScore: 75,
            standardScore: 105,
          },
        },
        optional: false,
        reliable: true,
      },
    },
    tasksDictionary: {
      swr: {
        publicName: 'Single Word Reading',
      },
    },
    longitudinalData: {},
    expanded: false,
  };

  beforeEach(() => {
    i18n = createI18n({
      legacy: false,
      locale: 'en',
      messages: {
        en: {
          scoreReports: {
            percentileTaskDescription: '{firstName} scored in the {percentile} on {taskName}, which {taskDescription}.',
          },
        },
      },
    });
  });

  describe('snapshots', () => {
    it('renders with single task', () => {
      const wrapper = mount(ScoreList, {
        props: defaultProps,
        global: {
          plugins: [i18n],
          stubs: {
            ScoreCard: true,
          },
        },
      });

      expect(wrapper.html()).toMatchSnapshot();
    });

    it('renders expanded state', () => {
      const wrapper = mount(ScoreList, {
        props: {
          ...defaultProps,
          expanded: true,
        },
        global: {
          plugins: [i18n],
          stubs: {
            ScoreCard: true,
          },
        },
      });

      expect(wrapper.html()).toMatchSnapshot();
    });

    it('renders with different student', () => {
      const wrapper = mount(ScoreList, {
        props: {
          ...defaultProps,
          studentFirstName: 'Sarah',
          studentGrade: '5',
        },
        global: {
          plugins: [i18n],
          stubs: {
            ScoreCard: true,
          },
        },
      });

      expect(wrapper.html()).toMatchSnapshot();
    });
  });

  describe('component behavior', () => {
    it('renders section header', () => {
      const wrapper = mount(ScoreList, {
        props: defaultProps,
        global: {
          plugins: [i18n],
          stubs: {
            ScoreCard: true,
          },
        },
      });

      expect(wrapper.text()).toContain('Detailed Assessment Results');
    });

    it('renders grid layout', () => {
      const wrapper = mount(ScoreList, {
        props: defaultProps,
        global: {
          plugins: [i18n],
          stubs: {
            ScoreCard: true,
          },
        },
      });

      const grid = wrapper.find('.grid');
      expect(grid.exists()).toBe(true);
      expect(grid.classes()).toContain('gap-2');
    });

    it('passes correct props to ScoreCard components', () => {
      const wrapper = mount(ScoreList, {
        props: defaultProps,
        global: {
          plugins: [i18n],
          stubs: {
            ScoreCard: true,
          },
        },
      });

      const scoreCards = wrapper.findAllComponents({ name: 'ScoreCard' });
      expect(scoreCards.length).toBeGreaterThan(0);
    });

    it('passes studentFirstName to ScoreCard', () => {
      const wrapper = mount(ScoreList, {
        props: defaultProps,
        global: {
          plugins: [i18n],
          stubs: {
            ScoreCard: true,
          },
        },
      });

      const scoreCard = wrapper.findComponent({ name: 'ScoreCard' });
      expect(scoreCard.exists()).toBe(true);
    });

    it('passes expanded prop to ScoreCard components', () => {
      const wrapper = mount(ScoreList, {
        props: {
          ...defaultProps,
          expanded: true,
        },
        global: {
          plugins: [i18n],
          stubs: {
            ScoreCard: true,
          },
        },
      });

      expect(wrapper.props('expanded')).toBe(true);
    });
  });
});
