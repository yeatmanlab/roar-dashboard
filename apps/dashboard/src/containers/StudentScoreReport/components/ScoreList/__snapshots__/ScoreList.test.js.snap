// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ScoreList.vue > snapshots > renders expanded state 1`] = `
"<section data-v-07dec570="" class="pt-4">
  <h2 data-v-07dec570="" class="text-2xl font-bold">Detailed Assessment Results</h2>
  <div data-v-07dec570="" class="grid gap-2 lg:grid-cols-2 xl:grid-cols-3">
    <score-card-stub data-v-07dec570="" publicname="Single Word Reading" scorelabel="Percentile Score" score="[object Object]" tags="[object Object]" valuetemplate="{value}%" scoretodisplay="percentileScore" studentfirstname="John" description="[object Object]" scoresarray="Raw Score,50,0,100,Percentile,75,0,100" expanded="true" longitudinaldata="" taskid="swr" studentgrade="3"></score-card-stub>
  </div>
</section>"
`;

exports[`ScoreList.vue > snapshots > renders with different student 1`] = `
"<section data-v-07dec570="" class="pt-4">
  <h2 data-v-07dec570="" class="text-2xl font-bold">Detailed Assessment Results</h2>
  <div data-v-07dec570="" class="grid gap-2 lg:grid-cols-2 xl:grid-cols-3">
    <score-card-stub data-v-07dec570="" publicname="Single Word Reading" scorelabel="Percentile Score" score="[object Object]" tags="[object Object]" valuetemplate="{value}%" scoretodisplay="percentileScore" studentfirstname="Sarah" description="[object Object]" scoresarray="Raw Score,50,0,100,Percentile,75,0,100" expanded="false" longitudinaldata="" taskid="swr" studentgrade="5"></score-card-stub>
  </div>
</section>"
`;

exports[`ScoreList.vue > snapshots > renders with single task 1`] = `
"<section data-v-07dec570="" class="pt-4">
  <h2 data-v-07dec570="" class="text-2xl font-bold">Detailed Assessment Results</h2>
  <div data-v-07dec570="" class="grid gap-2 lg:grid-cols-2 xl:grid-cols-3">
    <score-card-stub data-v-07dec570="" publicname="Single Word Reading" scorelabel="Percentile Score" score="[object Object]" tags="[object Object]" valuetemplate="{value}%" scoretodisplay="percentileScore" studentfirstname="John" description="[object Object]" scoresarray="Raw Score,50,0,100,Percentile,75,0,100" expanded="false" longitudinaldata="" taskid="swr" studentgrade="3"></score-card-stub>
  </div>
</section>"
`;
